Branch-SDK/Branch-SDK/BNCServerInterface.m

```diff
@@ -284,32 +284,58 @@ - (void) collectInstrumentationMetrics {
     [self.preferenceHelper clearInstrumentationDictionary];
     [self.preferenceHelper addInstrumentationDictionaryKey:brttKey value:lastRoundTripTime];
 }
+
 - (void)updateDeviceInfoToMutableDictionary:(NSMutableDictionary *)dict {
     BNCDeviceInfo *deviceInfo  = [BNCDeviceInfo getInstance];
 
-    NSString *hardwareId = [deviceInfo.hardwareId copy];
-    NSString *hardwareIdType = [deviceInfo.hardwareIdType copy];
+    NSString *hardwareId = deviceInfo.hardwareId;
+    NSString *hardwareIdType = deviceInfo.hardwareIdType;
     NSNumber *isRealHardwareId = @(deviceInfo.isRealHardwareId);
 
     if (hardwareId && hardwareIdType && isRealHardwareId) {
         dict[BRANCH_REQUEST_KEY_HARDWARE_ID] = hardwareId;
         dict[BRANCH_REQUEST_KEY_HARDWARE_ID_TYPE] = hardwareIdType;
         dict[BRANCH_REQUEST_KEY_IS_HARDWARE_ID_REAL] = isRealHardwareId;
     }
-    
-    [self safeSetValue:deviceInfo.vendorId forKey:BRANCH_REQUEST_KEY_IOS_VENDOR_ID onDict:dict];
-    [self safeSetValue:deviceInfo.brandName forKey:BRANCH_REQUEST_KEY_BRAND onDict:dict];
-    [self safeSetValue:deviceInfo.modelName forKey:BRANCH_REQUEST_KEY_MODEL onDict:dict];
-    [self safeSetValue:deviceInfo.osName forKey:BRANCH_REQUEST_KEY_OS onDict:dict];
-    [self safeSetValue:deviceInfo.osVersion forKey:BRANCH_REQUEST_KEY_OS_VERSION onDict:dict];
-    [self safeSetValue:deviceInfo.screenWidth forKey:BRANCH_REQUEST_KEY_SCREEN_WIDTH onDict:dict];
-    [self safeSetValue:deviceInfo.screenHeight forKey:BRANCH_REQUEST_KEY_SCREEN_HEIGHT onDict:dict];
-
-    [self safeSetValue:deviceInfo.browserUserAgent forKey:@"user_agent" onDict:dict];
-    [self safeSetValue:deviceInfo.country forKey:@"country" onDict:dict];
-    [self safeSetValue:deviceInfo.language forKey:@"language" onDict:dict];
-
-    dict[BRANCH_REQUEST_KEY_AD_TRACKING_ENABLED] = @(deviceInfo.isAdTrackingEnabled);
+
+    NSString *vendorId = deviceInfo.vendorId;
```

Comment:
Yes. (And it isn't better. I'd be inclined to revert it later after the problem is identified.) The stack trace consistently shows EXC_BAD_ACCESS from objc_msgSend, which this SO post says is due to a bad message recipient: https://stackoverflow.com/a/7684896/875725. It stands to reason, since obc_msgSend should not care about the args. When you have `[self safeSetValue:deviceInfo.vendorId etc:...]`, there are two messages sent in the same line. If the crash is consistent after this change, we can be certain it's the `BNCDeviceInfo` that's at fault rather than the `BNCServerInterface` (or vice versa).